boost.cmake

```diff
@@ -32,7 +32,7 @@ FUNCTION(SEARCH_FOR_BOOST_COMPONENT boost_python_name found)
 ENDFUNCTION(SEARCH_FOR_BOOST_COMPONENT boost_python_name found)
 
 IF(CMAKE_VERSION VERSION_LESS "3.12")
-  SET(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/boost ${CMAKE_MODULE_PATH})
+  SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/boost ${CMAKE_MODULE_PATH})
```

Comment:
`${CMAKE_CURRENT_LIST_DIR}/boost' will work correctly 100% of the time

In fact, in general it could be nice to pass through the jrl-cmakemodules to replace all assumptions of the `cmake` folder with `${CMAKE_CURRENT_LIST_DIR}`